packages/application-shell/src/components/setup-flop-flip-provider/setup-flop-flip-provider.tsx

```diff
@@ -54,23 +91,44 @@ export const SetupFlopFlipProvider = (props: Props) => {
 
   const adapterArgs = React.useMemo(
     () => ({
-      sdk: {
-        // Allow to overwrite the client ID, passed via the `additionalEnv` properties
-        // of the application config.
-        // This is mostly useful for internal usage on our staging environments.
-        clientSideId: props.ldClientSideId ?? ldClientSideIdProduction,
-      },
-      flags,
       user: {
         key: props.user?.id,
-        custom: omitEmpty<CustomLDUser, Partial<CustomLDUser>>({
-          id: props.user?.launchdarklyTrackingId,
-          project: props.projectKey,
-          team: props.user?.launchdarklyTrackingTeam,
-          group: props.user?.launchdarklyTrackingGroup,
-          subgroup: props.user?.launchdarklyTrackingSubgroup,
-          tenant: props.user?.launchdarklyTrackingTenant,
-        }),
+      },
+      memory: {
+        user: {
+          key: props.user?.id,
+        },
+      },
+      launchdarkly: {
+        sdk: {
+          // Allow to overwrite the client ID, passed via the `additionalEnv` properties
+          // of the application config.
+          // This is mostly useful for internal usage on our staging environments.
+          clientSideId: props.ldClientSideId ?? ldClientSideIdProduction,
+        },
+        flags,
+        user: {
+          key: props.user?.id,
+          custom: omitEmpty<CustomLDUser, Partial<CustomLDUser>>({
+            id: props.user?.launchdarklyTrackingId,
+            project: props.projectKey,
+            team: props.user?.launchdarklyTrackingTeam,
+            group: props.user?.launchdarklyTrackingGroup,
+            subgroup: props.user?.launchdarklyTrackingSubgroup,
+            tenant: props.user?.launchdarklyTrackingTenant,
+          }),
+        },
+      },
+      graphql: {
+        user: {
+          key: props.user?.id,
+        },
+        uri: `${getMcApiUrl()}/graphql`,
+        query: 'query AllFeatures { allFeatures { name value } }',
+        pollingInteralMs: 1000,
+        getRequestHeaders,
```

Comment:
It would be cool maybe to be able to pass the `apolloClient` here, to avoid having to configure the url, the headers, etc.

Reply:
Sorry, I have a PRs welcome for this ;)

Reply:
I can open a PR if you want, sure.

Reply:
I thought about it over night a bit more. What I don't like about it that `flopflip` "binds" to Apollo. I don't want it to be biased towards any GraphQL Client. Nor do I want to maintain and release breaking changes if Apollo does (as it would be a `peerDependency`). 

I somewhat like the generic nature of `fetch` and the `url`. The alternative would be to inverse control entirely. E.g. via a `fetchFlags: async () => TFlags`. Then a user could use `fetch` or any client.

Reply:
Yes I was also thinking about the generic approach.
I also don't want to bind the adapter to Apollo.

The main point is that the adapter should probably not be concerned about with making and configuring the HTTP request.

Reply:
So I guess it can then also be an `http-adapter` instead of something specific to graphql.

Reply:
Yeah in the end it could be renamed. Somehow it's not related to GraphQL after all.

Reply:
Then let's maybe do that before proceeding here?

Reply:
One of the few issues is that importing a GraphQl query results in a `DocumentNode` type. I don't know how to convert that into a string again or how Apollo does it when actually performing a fetch so far.

Reply:
Seems like `print` from graphql does that. I don't want to push that into the flopflip adpater. Also if it was an http-adapter. How would you proceed if you want to lint this tiny query while concerting the import to a string?

Reply:
If we have an `http-adapter`, the adapter does not concern itself with such things anymore. 

The adapter args would then be something like:

```js
http: {
  user: {
    key: props.user?.id,
  },
  execute,
  pollingIntervalMs: 1000,
  parseFlags,
  cacheIdentifier: 'local',
},
```

We have then the `execute` callback function, which does the data fetching.

Reply:
In the `execute` function, you then construct and execute the request as you want. So you can use apollo client and import the query as normal.

Reply:
Let me know if you want me to work on that in flopflip.

Reply:
Yeah. I agree. With an `http-adapter` the issue is gone. I like the abstraction too. If we ever need a GraphQl adapter it can compose or extend the http-adapter. Maybe one night I can work on it. I'll ping you for review.